Hivemind/Assets/Scripts/GameWorld.cs

```diff
@@ -45,43 +72,192 @@ public static ResourceNode FindNearestKnownResource(Vector3 antPosition, Resourc
+
+    public ResourceNode FindResourceNode(Guid guid)
+    {
+        return ResourceList.FirstOrDefault(resource => resource.myGuid == guid);
+    }
+
+    public void Save(string name = "QuickSave")
+    {
+        if (name == "QuickSave")
+        {
+            MyUnitController.UpdateEventText("QuickSaving...");
+        }
+        else
+        {
+            MyUnitController.UpdateEventText("Saving...");
+        }
+        SaveObject saveObject = new SaveObject
+        {
+            ResourceAmountsKeys = GameResources.GetResourceAmounts().Keys.ToList(),
+            ResourceAmountsValues = GameResources.GetResourceAmounts().Values.ToList(),
+            Resources = ResourceList,
+            Ants = AntList,
+            MindGroups = MyUnitController.MindGroupList.GetMindGroupList(),
+            BaseControllers = BaseControllerList
+        };
+        string json = saveObject.ToJson();
+        File.WriteAllText(GetSavePath() + $"/{name}.txt", json);
+        if (name == "QuickSave")
+        {
+            MyUnitController.UpdateEventText("QuickSave Complete!");
+        }
+        else
+        {
+            MyUnitController.UpdateEventText("Save Complete!");
+        }
     }
 
-    public static void AddNewResource(ResourceNode resource)
+    public string GetSavePath()
     {
-        UnknownResources.Add(resource);
+        string path = Application.dataPath + "/Saves";
+        if (!Directory.Exists(path))
+        {
+            Directory.CreateDirectory(path);
+        }
+        return path;
     }
 
-    public static void AddNewKnownResource(ResourceNode resource)
+    public void DeleteFile(string name)
     {
-        if (!KnownResources.Contains(resource))
+        if (File.Exists(GetSavePath() + $"/{name}.txt"))
         {
-            KnownResources.Add(resource);
-            UnknownResources.Remove(resource);
+            try
+            {
+                File.Delete(Application.dataPath + $"/Saves/{name}.txt");
+            }
+            catch
+            {
+                MyUnitController.UpdateEventText("Deleting save file failed!", Color.red);
+            }
         }
+        else
+        {
+            MyUnitController.UpdateEventText("Save file not found!", Color.red);
+        }
+    }
+
+    public void Load(string name = "QuickSave")
+    {
+        Instance.StartCoroutine(LoadEnumerator(name));
     }
 
-    //public static void CreateNewResource(int amount = 1)
-    //{
-    //    for (int i = 0; i < amount; i++)
-    //    {
-    //        GameObject newResource = GameObject.CreatePrimitive(PrimitiveType.Sphere);
-    //        newResource.transform.position = new Vector3(Random.Range(-50f, 50f), 0.5f, UnityEngine.Random.Range(-50f, 50f));
-    //        resources.Add(new ResourceNode(newResource.transform, newResource, UnityEngine.Random.Range(1, 5)));
-    //    }
-    //}
+    private IEnumerator LoadEnumerator(string name)
+    {
+        if (SceneManager.GetActiveScene().name != "Level_1")
```

Comment:
What if i want to open the load in the map. it will now switch towards level 1.

Reply:
Fixed by saving the levelname in the saveobject